chat-plugins/room-events.js

```diff
@@ -14,64 +14,84 @@ exports.commands = {
+
 			eventName = eventName.trim();
-			if (room.events[eventName]) return this.errorReply("An event with this name is already added to the events list.");
-			room.events[eventName] = {
-				desc: desc,
+			date = date.trim();
+			desc = desc.join('|').trim();
+
+			if (eventName.length > 50) return this.errorReply("Event names should not exceed 50 characters.");
+			if (date.length > 150) return this.errorReply("Event dates should not exceed 150 characters.");
+			if (desc.length > 1000) return this.errorReply("Event descriptions should not exceed 1000 characters.");
+
+			let eventId = toId(eventName);
+
+			room.events[eventId] = {
+				eventName: eventName,
 				date: date,
+				desc: desc,
 			};
 			this.privateModCommand(`(${user.name} added a roomevent titled "${eventName}".)`);
-			if (room.chatRoomData) {
-				room.chatRoomData.events = room.events;
-				Rooms.global.writeChatRoomData();
-			}
+
+			room.chatRoomData.events = room.events;
+			Rooms.global.writeChatRoomData();
 		},
 		remove: function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
 			if (!this.can('declare', null, room)) return false;
 			if (!room.events || Object.keys(room.events).length === 0) {
 				return this.errorReply("There are currently no planned upcoming events for this room to remove.");
 			}
 			if (!target) return this.errorReply("Usage: /roomevents remove [event name]");
-			target = target.trim();
-			if (!room.events[target]) return this.errorReply(`There is no such event named '${target}'.  Check spelling?`);
+			target = toId(target);
+			if (!room.events[target]) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
 			delete room.events[target];
 			this.privateModCommand(`(${user.name} removed a roomevent titled "${target}".)`);
-			if (room.chatRoomData) {
-				room.chatRoomData.events = room.events;
-				Rooms.global.writeChatRoomData();
+
+			room.chatRoomData.events = room.events;
+			Rooms.global.writeChatRoomData();
+		},
+		view: function (target, room, user) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!room.events || !Object.keys(room.events).length) {
+				return this.errorReply("There are currently no planned upcoming events for this room.");
 			}
+
+			if (!target) return this.errorReply("Usage: /roomevents view [event name]");
+			target = toId(target);
+			if (!room.events[target]) return this.errorReply(`There is no such event named '${target}'. Check spelling?`);
+
+			if (!this.runBroadcast()) return;
+			this.sendReplyBox(`<table border="1" cellspacing="0" cellpadding="3"><tr><td>${Chat.escapeHTML(room.events[target].eventName)}</td><td>${Chat.parseText(room.events[target].desc)}</td><td>${Chat.escapeHTML(room.events[target].date)}</td></tr></table>`);
```

Comment:
I think ``cellspacing`` and ``cellpadding`` aren't html5 compatible.  From what PS has been doing (avoiding ``<font>``, ``<center>``), maybe using CSS for this may the the "preferred" method.

Also, I'm pretty sure that this is susceptible to "constructor" crashes - haven't tested it, but it appears that way.